### DISCUSSION TITLE
tax_fix not resolving 'uncultured rumen bacterium 1-4'

### DISCUSSION THREAD

**Discussion by DomFS1506** [david-barnett/microViz#180] [General]
Hello, 

I am having issues with taxonomic classification of a set of 16s rRNA samples (murine faecal origin). Error message is 'problematic genus values detected in tax_table' due to convergent rows, but the issue appears to be misclassifying at the class level for 2 ASVs (see images attached). I have tried the tax_fix and tax_fix_interactive correction codes as shown but this does not resolve the misclassification. For this reason I able unable to plot any taxonomy. Any help resolving this would be much appreciated.

<img width="756" alt="Screenshot 2024-11-21 at 15 32 22" src="https://github.com/user-attachments/assets/0f1c71d6-3fc7-48b1-8d92-c84a88e95631">

<img width="1302" alt="Screenshot 2024-11-21 at 15 29 26" src="https://github.com/user-attachments/assets/2963d8b9-1b63-4fe3-adcd-301f724ace17">

Thanks,
Dominic

**Comment by david-barnett** [comment]
Hi Dominic,

Just to make sure, can you try this code, and copy-paste the full console code and output?

```R
# assuming your phyloseq is called ps (and that you have not already run tax fix on ps)

ps_fixed <- ps %>% tax_fix(unknowns = c("uncultured rumen bacterium"))

ps_fixed %>% tax_agg("Genus")

```

best,
David
